cmd/podman/common/create_opts.go

```diff
@@ -111,3 +121,276 @@ type ContainerCLIOpts struct {
+	// like with start. We believe this is just a difference in podman/compat
+	cliOpts := ContainerCLIOpts{
+		//Attach:            nil, // dont need?
+		Authfile:          "",
+		BlkIOWeight:       strconv.Itoa(int(cc.HostConfig.BlkioWeight)),
+		BlkIOWeightDevice: nil, // TODO
+		CapAdd:            append(capAdd, cc.HostConfig.CapAdd...),
+		CapDrop:           append(cappDrop, cc.HostConfig.CapDrop...),
+		CGroupParent:      cc.HostConfig.CgroupParent,
+		CIDFile:           cc.HostConfig.ContainerIDFile,
+		CPUPeriod:         uint64(cc.HostConfig.CPUPeriod),
+		CPUQuota:          cc.HostConfig.CPUQuota,
+		CPURTPeriod:       uint64(cc.HostConfig.CPURealtimePeriod),
+		CPURTRuntime:      cc.HostConfig.CPURealtimeRuntime,
+		CPUShares:         uint64(cc.HostConfig.CPUShares),
+		//CPUS:              0, // dont need?
+		CPUSetCPUs: cc.HostConfig.CpusetCpus,
+		CPUSetMems: cc.HostConfig.CpusetMems,
+		//Detach:            false, // dont need
+		//DetachKeys:        "",    // dont need
+		Devices:           devices,
+		DeviceCGroupRule:  nil,
+		DeviceReadBPs:     readBps,
+		DeviceReadIOPs:    readIops,
+		DeviceWriteBPs:    writeBps,
+		DeviceWriteIOPs:   writeIops,
+		Entrypoint:        &entrypoint,
+		Env:               cc.Config.Env,
+		Expose:            expose,
+		GroupAdd:          cc.HostConfig.GroupAdd,
+		Hostname:          cc.Config.Hostname,
+		ImageVolume:       "bind",
+		Init:              init,
+		Interactive:       cc.Config.OpenStdin,
+		IPC:               string(cc.HostConfig.IpcMode),
+		Label:             stringMaptoArray(cc.Config.Labels),
+		LogDriver:         cc.HostConfig.LogConfig.Type,
+		LogOptions:        stringMaptoArray(cc.HostConfig.LogConfig.Config),
+		Memory:            strconv.Itoa(int(cc.HostConfig.Memory)),
+		MemoryReservation: strconv.Itoa(int(cc.HostConfig.MemoryReservation)),
+		MemorySwap:        strconv.Itoa(int(cc.HostConfig.MemorySwap)),
+		Name:              cc.Name,
+		OOMScoreAdj:       cc.HostConfig.OomScoreAdj,
+		OverrideArch:      "",
+		OverrideOS:        "",
+		OverrideVariant:   "",
+		PID:               string(cc.HostConfig.PidMode),
+		PIDsLimit:         cc.HostConfig.PidsLimit,
+		Privileged:        cc.HostConfig.Privileged,
+		PublishAll:        cc.HostConfig.PublishAllPorts,
+		Quiet:             false,
+		ReadOnly:          cc.HostConfig.ReadonlyRootfs,
+		ReadOnlyTmpFS:     true, // podman default
+		Rm:                cc.HostConfig.AutoRemove,
+		SecurityOpt:       cc.HostConfig.SecurityOpt,
+		ShmSize:           strconv.Itoa(int(cc.HostConfig.ShmSize)),
+		StopSignal:        cc.Config.StopSignal,
+		StoreageOpt:       stringMaptoArray(cc.HostConfig.StorageOpt),
+		Sysctl:            stringMaptoArray(cc.HostConfig.Sysctls),
+		Systemd:           "true", // podman default
+		TmpFS:             stringMaptoArray(cc.HostConfig.Tmpfs),
+		TTY:               cc.Config.Tty,
+		//Ulimit:            cc.HostConfig.Ulimits,            // ask dan, no documented format
```

Comment:
What happens when you uncomment this?

Reply:
Most likely this corresponds to the argument values described in https://github.com/containers/buildah/blob/master/docs/buildah-bud.md.